### DISCUSSION TITLE
cookies not persisting in production when navigate between pages

### DISCUSSION THREAD

**Discussion by vinibgoulart** [vercel/next.js#45297] [Help]
cookies are setting in production, but when I reload the page or navitage to other them are removed

i'm using koajs in backend, and nextjs with relay in frontend

my app is on the same domain as the backend

how i doing the fetch: 
```ts
await fetch(myUrl, {
    method: 'POST',
    credentials: 'include',
    headers: {
      ...headers,
      'Content-Type': 'application/json',
    },
    body: JSON.stringify({
      query: params.text,
      variables,
    }),
  })
```

how i setting cookies:
```ts
const options = {
      httpOnly: true,
      secure: config.NODE_ENV !== 'development',
      sameSite: 'lax',
      path: '/',
    };

    context.koaContext.cookies.set('userToken', token, options);
```

# what is happening
### local
cookies are setting in backend and persisting in fronted

### production
cookies are setting in backend but not persisting in fronted
-- ----------------- ------------------ --------------- -------------------
![image](https://user-images.githubusercontent.com/88122830/214826714-81a83fb5-b7ee-4a43-a1c4-81db108434e2.png)
![image](https://user-images.githubusercontent.com/88122830/214826966-3cc9be11-f091-4712-ae59-1edc180c3eb6.png)
-- ----------------- ------------------ --------------- -------------------
cookies are setted, but after reload page or navigate:
![image](https://user-images.githubusercontent.com/88122830/214827060-58663508-eb4f-4242-bbbd-39f59ad2ab8d.png)

**Comment by icyJoseph** [comment]
Try to set an explicit time for your token, right now it has session as expiry time, so it is no surprise that it'll disappear.
